### DISCUSSION TITLE
Window on another workspace force focus attention

### DISCUSSION THREAD

**Discussion by karot9100** [i3/i3#5656] [Q&A]
Is it possible to let an app take focus, even if you are on another workspace?

in this case i'm talking about RuneLite. There is an option inside the app that sends notifications if you for example go idle in the game. You can also let it force focus on the program. This works on windows, not sure if it worked on another distro/wm.

I tried focus_on_window_activation but it does not work. Might misunderstand what this does tho.

**Comment by okraits** [comment]
I'm not sure if I understand the whole issue but I try to give a general answer.

You could use the IPC interface (https://i3wm.org/docs/ipc.html) to subscribe to the event you are looking for (focus event of a specific window), and then trigger suitable actions (switch to the workspace with the window and focus the window).

**Comment by orestisfl** [comment]
Well, in theory you should be able to add
```
for_window [urgent=latest class=RuneLite] focus
```
in your config but that doesn't work, I've opened an issue & PR: https://github.com/i3/i3/issues/5658

**Reply by BirdeeHub** [reply]
here is a script where i use i3-msg to forcibly change focus and move workspaces around between monitors. There should be something in here that works for you. 2 scripts actually. Learn from this and use i3-msg to do it, and find a way to trigger it on window open, either udev or xsession events. you can get all the workspace info from i3-msg, and do the force focus, but you may have to use some of that info to query something else to get the ACTUAL window within the pane

ipc as the other commenter said may be better for the actual window, but the command to focus a workspace is in here at least along with some ideas

https://github.com/BirdeeHub/i3autoXrandrMemory

**Comment by karot9100** [comment]
Thanks for the info guys, i'll get to some testing with my limited skills :D

**Comment by prosoitos** [comment]
I am having a new issue somewhat related to this. I don't think that #5659 triggered my problem as I think that it started a little earlier, although I am not sure which is why I am posting here.

For years, when I would open a text file from ranger in one workspace into Emacs in another workspace, the focus would shift from the ranger workspace into the Emacs workspace automatically. It was very convenient because I would automatically be into my newly open file.

Recently, something broke this behaviour. When I open a file from ranger in a workspace into Emacs in another workspace, the focus remain in the ranger workspace and I have to manually switch focus to my Emacs workspace before I can edit the file.

Note that this new behaviour does not happen if Emacs is in the same workspace as ranger (so the behaviour is fine between windows in one workspace. In that case, the focus jumps to Emacs as it always had. It is only between workspaces that something broke the jump in focus).

I tried to add `for_window [urgent=latest class=Emacs] focus` to see whether that would fix this new behaviour but to no effect.

Any idea what recent change created this?

Thank you!

Note: happy to open a new issue if that would be better.

**Comment by prosoitos** [comment]
I found the cause of the problem:

I had the setting `focus_on_window_activation urgent`. Reverting this to the default fixes things.

This is a new behaviour however under the setting `focus_on_window_activation urgent` as I had had that setting for a long time (as it helps in something else in my workflow). So something did change. However, the new behaviour, while inconvenient in my case, might be what one would expect under this option, so the change might have been a fix rather than something getting broken. It just happens not to work well for me, but my comments should probably be ignored I guess...

Thank you!
